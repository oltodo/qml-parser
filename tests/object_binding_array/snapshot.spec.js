const { parseFile } = require("../../bin");

it("renders correctly", () => {
  const ast = parseFile(`${__dirname}/entry.qml`);

  expect(ast).toMatchSnapshot();
});
