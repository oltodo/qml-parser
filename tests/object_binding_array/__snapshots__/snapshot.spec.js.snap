// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "children": Array [
    Object {
      "identifier": "QtQuick",
      "kind": "Import",
      "loc": Object {
        "end": Object {
          "column": 18,
          "line": 1,
          "offset": 18,
        },
        "start": Object {
          "column": 0,
          "line": 1,
          "offset": 0,
        },
      },
      "version": "2.0",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "identifier": "name",
                  "kind": "Attribute",
                  "loc": Object {
                    "end": Object {
                      "column": 17,
                      "line": 7,
                      "offset": 74,
                    },
                    "start": Object {
                      "column": 6,
                      "line": 7,
                      "offset": 63,
                    },
                  },
                  "value": Object {
                    "kind": "JavascriptValue",
                    "loc": Object {
                      "end": Object {
                        "column": 17,
                        "line": 7,
                        "offset": 74,
                      },
                      "start": Object {
                        "column": 12,
                        "line": 7,
                        "offset": 69,
                      },
                    },
                    "object": false,
                    "value": "\\"foo\\"",
                  },
                },
              ],
              "identifier": "State",
              "kind": "ObjectDefinition",
              "loc": Object {
                "end": Object {
                  "column": 5,
                  "line": 8,
                  "offset": 81,
                },
                "start": Object {
                  "column": 4,
                  "line": 6,
                  "offset": 49,
                },
              },
            },
            Object {
              "children": Array [
                Object {
                  "identifier": "name",
                  "kind": "Attribute",
                  "loc": Object {
                    "end": Object {
                      "column": 17,
                      "line": 10,
                      "offset": 112,
                    },
                    "start": Object {
                      "column": 6,
                      "line": 10,
                      "offset": 101,
                    },
                  },
                  "value": Object {
                    "kind": "JavascriptValue",
                    "loc": Object {
                      "end": Object {
                        "column": 17,
                        "line": 10,
                        "offset": 112,
                      },
                      "start": Object {
                        "column": 12,
                        "line": 10,
                        "offset": 107,
                      },
                    },
                    "object": false,
                    "value": "\\"bar\\"",
                  },
                },
              ],
              "identifier": "State",
              "kind": "ObjectDefinition",
              "loc": Object {
                "end": Object {
                  "column": 5,
                  "line": 11,
                  "offset": 119,
                },
                "start": Object {
                  "column": 4,
                  "line": 9,
                  "offset": 87,
                },
              },
            },
          ],
          "identifier": "states",
          "kind": "ArrayBinding",
          "loc": Object {
            "end": Object {
              "column": 3,
              "line": 12,
              "offset": 123,
            },
            "start": Object {
              "column": 10,
              "line": 5,
              "offset": 43,
            },
          },
        },
      ],
      "identifier": "Rectangle",
      "kind": "ObjectDefinition",
      "loc": Object {
        "end": Object {
          "column": 1,
          "line": 14,
          "offset": 126,
        },
        "start": Object {
          "column": 0,
          "line": 3,
          "offset": 20,
        },
      },
    },
  ],
  "comments": Array [],
  "kind": "Program",
  "loc": Object {
    "end": Object {
      "column": 0,
      "line": 15,
      "offset": 127,
    },
    "start": Object {
      "column": 0,
      "line": 1,
      "offset": 0,
    },
  },
}
`;
